### PR TITLE
Make shuttle walls/windows constructable/deconstructable

### PR DESCRIPTION
**Pull request by IamVelcroboy**
## About the PR
- Makes shuttle walls(base and diagonal) and windows deconstructble/constructable. 
- Decon difficulty is high to prevent tomfoolery
- Adds an interior shuttle wall that is less strong
- Adds missing break sound to reinforced walls
- Makes shuttle windows repairable 
- Tested downstream since December

**Media**
image
image
Not sure if you need any more media for this. Just ask if you do.

- [x] I have added screenshots/videos to this PR showcasing its changes in-game, **or** this PR does not require an in-game showcase

**Changelog**

:cl: Velcroboy
- tweak: Shuttle walls and windows can now be constructed.

### CODE REVIEW

**Comment by ChilbroBaggins** [comment]
Unrelated test failure as far as I can tell.

File: Content.Shared/Wall/ShuttleWallVisuals.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Wall
+{
+    [Serializable, NetSerializable]
+    public enum ShuttleWallVisuals
+    {
+        DeconstructionStage,
+    }
+}
```
Comment (metalgearsloth) [inline Content.Shared/Wall/ShuttleWallVisuals.cs:10]: Does this need a new byte over reinforcedwallvisuals or the likes?

Also file-scoped namespace.

File: Resources/Prototypes/Entities/Structures/Walls/walls.yml [hunk 2]
```diff
@@ -640,10 +645,11 @@
     key: walls
     base: silver
 
+#shuttle walls
 - type: entity
   id: WallShuttleDiagonal
   name: shuttle wall
-  suffix: diagonal
+  suffix: Diagonal
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Structures/Walls/walls.yml:652]: should be lowercase still.

File: Resources/Prototypes/Entities/Structures/Walls/walls.yml [hunk 3]
```diff
@@ -657,6 +663,7 @@
     tags:
     - Wall
   - type: Sprite
+    netsync: false
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Structures/Walls/walls.yml:666]: this is irrelevant.

File: Resources/Prototypes/Recipes/Construction/Graphs/structures/girder.yml [hunk 4]
```diff
@@ -7,7 +7,7 @@
         - to: girder
           completed:
             - !type:SnapToGrid
-              southRotation: true
+              southRotation: false
```
Comment (metalgearsloth) [inline Resources/Prototypes/Recipes/Construction/Graphs/structures/girder.yml:10]: Girders should still be south-facing.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Recipes/Construction/Graphs/structures/girder.yml (hunk 4 above)
Comment (ChilbroBaggins) [reply Resources/Prototypes/Recipes/Construction/Graphs/structures/girder.yml:10]: This one is necessary for constructing the diagonal walls. If it snaps to South, the walls will only construct in South position.

File: Resources/Prototypes/Entities/Structures/Walls/walls.yml (hunk 2 above)
Comment (ChilbroBaggins) [reply Resources/Prototypes/Entities/Structures/Walls/walls.yml:652]: All the others are upper-case. So, this was for consistency.

File: Content.Shared/Wall/ShuttleWallVisuals.cs (hunk 1 above)
Comment (ChilbroBaggins) [reply Content.Shared/Wall/ShuttleWallVisuals.cs]: >Does this need a new byte over reinforcedwallvisuals or the likes?

If you have a better suggestion, I'm open to it? This just worked....

Fixed namespace.

File: Resources/Prototypes/Recipes/Construction/Graphs/structures/girder.yml (hunk 4 above)
Comment (ChilbroBaggins) [reply Resources/Prototypes/Recipes/Construction/Graphs/structures/girder.yml:10]: Ok, with further testing, I might have been wrong. Going to double test. But yeah, seems this is unnecessary.

**Comment by ChilbroBaggins** [comment]
Ok, I think that addresses everything. Other than the casing. See my explanation.

File: Resources/Prototypes/Entities/Structures/Walls/girders.yml [hunk 5]
```diff
@@ -6,7 +6,7 @@
   components:
   - type: Transform
     anchored: true
-    noRot: true
+    noRot: false
```
Comment (Partmedia) [inline Resources/Prototypes/Entities/Structures/Walls/girders.yml:9]: BaseStructureDynamic is parented by many many items, so this would likely cause a lot of breakage. If this is important for your changes, add noRot to the new prototypes that require them.

File: Resources/Prototypes/Entities/Structures/Walls/walls.yml [hunk 6]
```diff
@@ -705,51 +711,69 @@
...
+    sprite: Structures/Walls/shuttleinterior.rsi
+  - type: Icon
+    sprite: Structures/Walls/shuttleinterior.rsi
+  - type: Construction
+    graph: Girder
+    node: shuttleInteriorWall
   - type: IconSmooth
     key: walls
     base: state
+  - type: Reflect
```
Comment (Partmedia) [inline Resources/Prototypes/Entities/Structures/Walls/walls.yml:775]: This might lead to some funny bouncing especially in small shuttles, but I'm not entirely opposed.

File: Resources/Prototypes/Recipes/Construction/Graphs/structures/girder.yml [hunk 7]
```diff
@@ -43,7 +43,7 @@
         - to: reinforcedGirder
           completed:
             - !type:SnapToGrid
-              southRotation: true
+              southRotation: false
```
Comment (Partmedia) [inline Resources/Prototypes/Recipes/Construction/Graphs/structures/girder.yml:46]: Why was this changed? This will affect ordinary walls, too. Why do shuttle walls need special-casing?

File: Resources/Prototypes/Recipes/Construction/Graphs/structures/girder.yml [hunk 8]
```diff
@@ -145,6 +145,22 @@
               amount: 2
               doAfter: 2
 
+        - to: shuttleInteriorWall
+          completed:
+            - !type:SnapToGrid
+              southRotation: true
```
Comment (Partmedia) [inline Resources/Prototypes/Recipes/Construction/Graphs/structures/girder.yml:151]: Is this needed because the girder SnapToGrid was removed?

File: Resources/Prototypes/Recipes/Construction/Graphs/structures/window.yml [hunk 9]
```diff
@@ -186,3 +194,34 @@
...
+          completed:
+            - !type:SpawnPrototype
+              prototype: SheetRGlass1
+              amount: 2
+            - !type:SpawnPrototype
+              prototype: SheetPlasteel1
+              amount: 2
+            - !type:DeleteEntity {}
+          steps:
+          #To-Do: Add deconstruct visuals
```
Comment (Partmedia) [inline Resources/Prototypes/Recipes/Construction/Graphs/structures/window.yml:211]: Is this still a TODO?

File: Resources/Prototypes/Recipes/Construction/Graphs/structures/girder.yml (hunk 8 above)
Comment (IamVelcroboy) [reply Resources/Prototypes/Recipes/Construction/Graphs/structures/girder.yml:151]: Honestly, that's just how they are all set up.

File: Resources/Prototypes/Recipes/Construction/Graphs/structures/girder.yml (hunk 7 above)
Comment (IamVelcroboy) [reply Resources/Prototypes/Recipes/Construction/Graphs/structures/girder.yml:46]: It's the diagonal walls that need this. If they snap false, it un-rotates the construction ghost. 
image
image

W/ southRotation set to false:
image

**Comment by Partmedia** [comment]
Shuttle wall construction is broken:
image

```
[ERRO] runtime: Caught exception in "GameLoop Tick"
Robust.Shared.GameObjects.EntityCreationException: Attempted to spawn an entity with an invalid prototype: WallShuttle
```

**Comment by IamVelcroboy** [comment]
> Shuttle wall construction is broken: image
> 
> ```
> [ERRO] runtime: Caught exception in "GameLoop Tick"
> Robust.Shared.GameObjects.EntityCreationException: Attempted to spawn an entity with an invalid prototype: WallShuttle
> ```

Something must have happened with the last merge conflict fix. I'm not at a computer today but I'll look into it tomorrow.

**Comment by IamVelcroboy** [comment]
> Shuttle wall construction is broken: 

Hey, I'm testing this right now and was able to construct the wall with no issues, no errors.
image

**Comment by Partmedia** [comment]
My apologies, something must have been going wrong in my testing environment. It looks like things are working. Thanks for investigating.

**Comment by Partmedia** [comment]
Thank you for your contribution. Your work is appreciated. Sorry for the long wait in review.

**Comment by IamVelcroboy** [comment]
> Thank you for your contribution. Your work is appreciated. Sorry for the long wait in review.

Thanks! And no worries!
